src/main/java/duke/application/Parser.java

```diff
@@ -0,0 +1,288 @@
+
+
+/**
+ * The Parser class is responsible for parsing user input and executing corresponding commands.
+ * It handles various commands such as listing tasks, marking/unmarking tasks, adding todos,
+ * deadlines, and events, removing tasks, and exiting the application.
+ */
+public class Parser {
+
+    /**
+     * Parses the user input into an array of strings, separating the command and the remaining input.
+     *
+     * @param input The user input to be parsed.
+     * @return An array containing the command and the remaining input.
+     */
+    public static String[] parseInput(String input) {
+        return input.split(" ", 2);
+    }
+
+    /**
+     * Handles the execution of commands based on the parsed user input.
+     *
+     * @param input    The user input to be processed.
+     * @param taskList The task list to be manipulated.
+     */
+    public void handleCommand(String input, TaskList taskList) {
+        String[] parsedInput = Parser.parseInput(input);
+
+        if ("list".equalsIgnoreCase(input)) {
+            Ui.showTaskList(taskList.getTaskList());
+
+        } else if ("bye".equalsIgnoreCase(input)) {
+            Ui.showByeMessage();
+            Storage.saveTasks(taskList.getTaskList());
+            System.exit(0);
+        } else if (parsedInput[0].equalsIgnoreCase("mark") || parsedInput[0].equalsIgnoreCase("unmark")) {
+            markingHandler(input, taskList);
+        } else if (parsedInput[0].equalsIgnoreCase("deadline")) {
+            if (isDeadlineInput(input)) {
+                handleDeadlines(input, taskList);
+            } else {
+                Ui.showErrorMessage("Please complete your request by specifying the details of the task!");
+            }
+        } else if (parsedInput[0].equalsIgnoreCase("todo")) {
+            if (isTodoInput(input)) {
+                handleTodos(input, taskList);
+            } else {
+                Ui.showErrorMessage("Please complete your request by specifying the details of the task!");
+            }
+        } else if (parsedInput[0].equalsIgnoreCase("event")) {
+            if (isEventInput(input)) {
+                handleEvents(input, taskList);
+            } else {
+                Ui.showErrorMessage("Please complete your request by specifying the details of the task!");
+            }
+        } else if (parsedInput[0].equalsIgnoreCase("delete")) {
+            handleRemove(input, taskList);
+        } else if (parsedInput[0].equalsIgnoreCase("find")) {
+            findItems(input, taskList);
+        } else {
+            Ui.showErrorMessage("I'm sorry, I don't understand! Please type your request again.");
+        }
+    }
+
+    /**
+     * Handles the marking or unmarking of tasks based on user input.
+     *
+     * @param input    The user input specifying the task number and the action.
+     * @param taskList The task list to be manipulated.
+     */
+    public static void markingHandler(String input, TaskList taskList) {
```

Comment:
As most of your other methods follow the verb-noun naming (e.g. handleTodos), and as this method indeed primarily handles the marking/unmarking of tasks, maybe can consider changing the name to "handleMarking"?